### PR TITLE
milestone 1: header

### PR DESCRIPTION
**Pull request by Emmy-github-webdev**
## Portfolio milestone 1: header

In this milestone 1 :
- Add toolbar for my portfolio
- Add headline section for my portfolio
- Setup a linter for GitHub
- Implement Github flow

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  <link rel="stylesheet" href="assets/css/styles.css">
+  <title>Welcome: I'm a Software Developer!</title>
+</head>
+<body>
+  <div class="homepage">
+    <header>
+      <nav>
+        <div class="nav-container">
+          <div class="logo">
+            <img src="assets/images/mobile/My-Logo.svg" alt="Logo">
```
Comment (carlos-ssh) [inline index.html:19]: - Your logo should be nested in a `` <a> `` tag.

File: assets/css/styles.css [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+
+nav {
+  position: absolute;
+  width: 414px;
+  height: 72px;
+  left: 0;
+}
+
+.nav-container {
+  position: absolute;
```
Comment (carlos-ssh) [inline assets/css/styles.css:40]: - Your content should be aligned with flexbox.

File: index.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          <li><img src="assets/images/mobile/Ellipse12.svg" class="ellipse12" alt="image"></li>
+          <li><img src="assets/images/mobile/Rectangle6.svg" class="rectangle6" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse11.svg" class="ellipse11" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse8.svg" class="ellipse8" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse14.svg" class="ellipse14" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse15.svg" class="ellipse15" alt="image"></li>
+          <li><img src="assets/images/mobile/Rectangle4.svg" class="rectangle4" alt="image"></li>
+        </ul>
+
+        <div class="headline">
```
Comment (carlos-ssh) [inline index.html:43]: - Your ``<div class="headline">`` should be inside a ``<section>``  tag and outside of your ``<header>``.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            <div class="social-media-text">
+              <h2>LET’S CONNECT</h2>
+            </div>
+            <div class="social-media-image">
+              <ul>
+                <li><a href="#"><img src="assets/images/mobile/twitter.svg" class="twitter-icon" alt=""></a></li>
+                <li><a href="#"><img src="assets/images/mobile/Linkedin-icon.svg" class="linkedin-icon" alt=""></a></li>
+                <li><a href="#"><img src="assets/images/mobile/m.svg" class="m-icon" alt=""></a></li>
+                <li><a href="#"><img src="assets/images/mobile/github.svg" class="github-icon" alt=""></a></li>
+                <li><a href="#"><img src="assets/images/mobile/finger.svg" class="finger-icon" alt=""></a></li>
```
Comment (carlos-ssh) [inline index.html:57]: - Kindly fill your ``alt=""`` parameters.

File: assets/css/styles.css [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+.ellipse11 {
+  position: absolute;
+  width: 38.5px;
+  height: 19.25px;
+  left: -20px;
+  top: 131.47px;
+  transform: rotate(-156.88deg);
+}
+
+.ellipse8 {
```
Comment (carlos-ssh) [inline assets/css/styles.css:197]: [OPTIONAL]
- Your ``.ellipse8`` and ``.ellipse15`` contains similar parameters you could improve your code using DRY _(Optimized code and less lines)._

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hi @Emmy-github-webdev, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: assets/css/styles.css (hunk 5 above)
Comment (Emmy-github-webdev) [reply assets/css/styles.css:197]: I will appreciate a hint to apply this to both .ellipse8 and .ellipse15 from a single point. They are both different images. Thank you.

Comment (carlos-ssh) [reply assets/css/styles.css:198]: You can found more information about that [here](https://vanseodesign.com/css/dry-principles/).
- This is a way to DRY your code:
```
.ellipse8
.ellipse15  {
 position: absolute;
  width: 15px;
  height: 15px;
  background: #fff;
  box-sizing: border-box;
}

.ellipse8 {
  left: 214px;
  top: 110px;
}

.ellipse15  {
  left: 53px;
  top: 680px;
}
 ```
 
 _I hope that this information going to be helpful for you._

File: index.html [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+          <h1>I’m Emmanuel Glad to see you!</h1>
+          <p>I’m a software developer! I can help you build a product , feature or website Look through some of my work and experience! If you like what you see and have a project you need coded, don’t hestiate to contact me.</p>
+
+          <div class="social-media">
+            <div class="social-media-text">
+              <h2>LET’S CONNECT</h2>
+            </div>
+            <div class="social-media-image">
+              <ul>
+                <li><a href="#"><img src="assets/images/mobile/twitter.svg" class="twitter-icon" alt=""></a></li>
```
Comment (carlos-ssh) [inline index.html]: - Kindly fill your ``alt=""`` parameters.

File: index.html [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+          <li><img src="assets/images/mobile/Rectangle6.svg" class="rectangle6" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse11.svg" class="ellipse11" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse8.svg" class="ellipse8" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse14.svg" class="ellipse14" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse15.svg" class="ellipse15" alt="image"></li>
+          <li><img src="assets/images/mobile/Rectangle4.svg" class="rectangle4" alt="image"></li>
+        </ul>
+
+        <div class="headline">
+          <h1>I’m Emmanuel Glad to see you!</h1>
```
Comment (carlos-ssh) [inline index.html:47]: **Kindly follow the changes requested on the last review.**

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hi @Emmy-github-webdev , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: assets/css/styles.css (hunk 5 above)
Comment (Emmy-github-webdev) [reply assets/css/styles.css:198]: Thank you

File: index.html [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+          <li><img src="assets/images//mobile//Ellipse4.svg" class="ellipse4" alt="ellipse4"></li>
+          <li><img src="assets/images/mobile/Ellipse3.svg" class="ellipse3" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse13.svg" class="ellipse13" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse12.svg" class="ellipse12" alt="image"></li>
+          <li><img src="assets/images/mobile/Rectangle6.svg" class="rectangle6" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse11.svg" class="ellipse11" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse8.svg" class="ellipse8" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse14.svg" class="ellipse14" alt="image"></li>
+          <li><img src="assets/images/mobile/Ellipse15.svg" class="ellipse15" alt="image"></li>
+          <li><img src="assets/images/mobile/Rectangle4.svg" class="rectangle4" alt="image"></li>
```
Comment (dasileker) [inline index.html:43]: - [ ] This **svg** that you added should  be as a background, please fix.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Emmy-github-webdev,** 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Temmarie** [comment]
Hi @Emmy-github-webdev 👋, 

_This is an additional review as a result of the second opinion request._

Thanks for reaching out to us for a second opinion, you complained about the last code reviewer requested a change not mentioned in the first 2 reviews.
Since the first 2 reviews didn't mention the last change required it should be marked as optional. 
As this was the only reason why your project was not approved, it will be done as soon as possible since you've exhausted your code review request. However, the change is recommended for you to make before you merge, as the review is valid although it should have been marked as optional. 

I hope this helps to resolve the issue. 

Happy coding and keep rocking !! 😎

**Comment by Emmy-github-webdev** [comment]
Hello Grace,

Thank you for your response.

It is still showing incomplete on my dashboard.

[image: image.png]

Can I continue with the next project?

My best regards,

**Pull request merged**